posthog/caching/warming.py

```diff
@@ -126,13 +127,18 @@ def schedule_warming_for_teams_task():
     max_retries=3,
 )
 def warm_insight_cache_task(insight_id: int, dashboard_id: Optional[int]):
-    insight = Insight.objects.get(pk=insight_id)
+    try:
+        insight = Insight.objects.get(pk=insight_id)
+    except Insight.DoesNotExist:
+        logger.info(f"Warming insight cache failed 404 insight not found: {insight_id}")
+        return
+
     dashboard = None
 
     tag_queries(team_id=insight.team_id, insight_id=insight.pk, trigger="warmingV2")
     if dashboard_id:
         tag_queries(dashboard_id=dashboard_id)
-        dashboard = insight.dashboards.get(pk=dashboard_id)
+        dashboard = insight.dashboards.filter(pk=dashboard_id).first()
```

Comment:
why do we need to change this? Insights can be added to a dashboard multiple times?

Reply:
Get throws an exception if it doesn’t exist, but we were expecting a none, which was leading to sentry errors